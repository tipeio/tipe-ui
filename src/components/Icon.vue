<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
