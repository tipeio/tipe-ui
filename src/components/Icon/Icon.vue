<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
