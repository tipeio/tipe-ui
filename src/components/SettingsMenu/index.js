export { default } from './SettingsMenu.vue'
