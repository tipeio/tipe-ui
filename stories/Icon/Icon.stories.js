import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
