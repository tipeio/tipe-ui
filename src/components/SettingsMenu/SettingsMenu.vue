<template>
  <ul>
    <li><div class="icon"><icon icon="wrench"/></div><p class="label">General</p></li>
    <li><div class="icon"><icon icon="users"/></div><p class="label">Members</p></li>
    <li><div class="icon"><icon icon="key"/></div><p class="label">API Key</p></li>
    <li><div class="icon"><icon icon="branch"/></div><p class="label">Webhooks</p></li>
    <li><div class="icon"><icon icon="credit-card"/></div><p class="label">Billing</p></li>
  </ul>
</template>

<script>
import Icon from '../Icon'

export default {
  name: 'SettingsMenu',
  components: {
    Icon
  }
}
</script>

<style lang="postcss">
ul {
  height: 305px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 191px;
  justify-content: space-evenly;
  margin: 0;
  padding: 0;
}
li {
  display: flex;
  align-items: center;
  height: 2.5625rem;
  list-style-type: none;
  justify-content: flex-start;
  width: 100%;

  &:hover {
    background-color: var(--purple);
    border-radius: 4px;

    & .label {
      color: #fff;
    }
  }

  & .icon {
    display: flex;
    justify-content: center;
    flex: 0 0 4.5rem;
  }

  & .label {
    display: flex;
    flex: 1 1 auto;
    color: var(--darkPurple);
    font-size: 0.8125rem;
    margin: 0;
    text-align: center;
  }
}
</style>
