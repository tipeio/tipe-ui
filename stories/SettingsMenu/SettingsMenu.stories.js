import { storiesOf } from '@storybook/vue'

import SettingsMenu from '../../src/components/SettingsMenu'

storiesOf('Settings Menu', module).add('settings menu', () => ({
  components: { SettingsMenu },
  template: '<settings-menu/>'
}))
