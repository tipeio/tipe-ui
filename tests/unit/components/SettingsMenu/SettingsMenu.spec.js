import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import SettingsMenu from '@/components/SettingsMenu'
import Icon from '@/components/Icon'

describe('SettingsMenu.vue', () => {
  it('renders', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(SettingsMenu)
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
  it('renders the correct number of list items', () => {
    const wrapper = shallowMount(SettingsMenu)
    const liArr = wrapper.findAll('li')
    expect(liArr).toHaveLength(5)
  })
  it('renders list items with icons', () => {
    const wrapper = shallowMount(SettingsMenu)
    const liArr = wrapper.findAll('li')
    expect(
      liArr
        .at(0)
        .find(Icon)
        .exists()
    ).toBe(true)
    expect(
      liArr
        .at(1)
        .find(Icon)
        .exists()
    ).toBe(true)
    expect(
      liArr
        .at(2)
        .find(Icon)
        .exists()
    ).toBe(true)
    expect(
      liArr
        .at(3)
        .find(Icon)
        .exists()
    ).toBe(true)
    expect(
      liArr
        .at(4)
        .find(Icon)
        .exists()
    ).toBe(true)
  })
})
