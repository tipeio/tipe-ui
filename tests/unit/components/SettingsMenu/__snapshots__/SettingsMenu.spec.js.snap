// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsMenu.vue renders 1`] = `
<ul>
  <li class="first">
    <div class="icon">
      <Icon-stub icon="wrench" height="1rem" width="1rem" color="#000" __iconLoader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}"></Icon-stub>
    </div>
    <p class="label">General</p>
  </li>
  <li>
    <div class="icon">
      <Icon-stub icon="user" height="1rem" width="1rem" color="#000" __iconLoader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}"></Icon-stub>
    </div>
    <p class="label">Members</p>
  </li>
  <li>
    <div class="icon">
      <Icon-stub icon="key" height="1rem" width="1rem" color="#000" __iconLoader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}"></Icon-stub>
    </div>
    <p class="label">API Key</p>
  </li>
  <li>
    <div class="icon">
      <Icon-stub icon="flow" height="1rem" width="1rem" color="#000" __iconLoader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}"></Icon-stub>
    </div>
    <p class="label">Webhooks</p>
  </li>
  <li class="last">
    <div class="icon">
      <Icon-stub icon="credit-card" height="1rem" width="1rem" color="#000" __iconLoader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}"></Icon-stub>
    </div>
    <p class="label">Billing</p>
  </li>
</ul>
`;
